import Head from 'next/head';
import styles from '../styles/Home.module.css';
import { useRouter } from 'next/router';
import Link from 'next/link';
import { Box, AppBar, Toolbar, Typography, Button, Card, CardContent } from '@mui/material';

export default function ReadPaper() {

 const router = useRouter();
 const showPaper1 = router.query.showPaper1;
 const showPaper2 = router.query.showPaper2;
 const showPaper3 = router.query.showPaper3;

 const goHome = () => {
  window.location.href = "http://letsgobiotech.com/";
 }

 const seeAboutUs = () => {
  window.location.href= "http://letsgobiotech.com/about-us";
 }

 //Paper 2
 const paperAbstract2 = "Analysis of cell-free DNA (cfDNA) in the blood has shown promise for monitoring a number of biological processes, but as circulating DNA predominantly originates from blood cells and other healthy tissues, the sensitivity of the approach for detecting disease may be reduced by these background signals. Writing in Nature Biotechnology, Esfahani et al.1 show that the expression level of genes captured in cfDNA samples can be predicted using the size variability of the fragments covering the transcription start site of a gene. Active regions of the transcription start site tend to be depleted of nucleosomes, leading to more-random cleavage patterns and higher variability in DNA fragment length. The authors introduce a metric, which they term promoter fragmentation entropy (PFE), that is strongly correlated with RNA expression level and shows good sensitivity for detection and classification of cancer from cfDNA (Fig. 1). Such a metric adds to our armamentarium of fragmentomic features of cfDNA for scrutinizing the cancer signal in blood circulation, augmenting current efforts in developing noninvasive biomarkers for cancer detection and monitoring.";
 // Paper 1
 const paperAbstract1 = "RNA-guided systems, which employ complementarity between a guide RNA and target nucleic acid sequences for recognition of genetic elements, play a central role in biological processes in both prokaryotes and eukaryotes. For example, the prokaryotic CRISPR-Cas systems provide adaptive immunity for bacteria and archaea against foreign genetic elements. Cas effectors, like Cas9 and Cas12, perform guide RNA-dependent DNA cleavage1. Although a few eukaryotic RNA-guided systems have been studied so far, such as RNA interference2 and ribosomal RNA modification3, it remains unclear whether eukaryotes have RNA-guided endonucleases. Recently, a new class of prokaryotic RNA-guided system (termed OMEGA) was reported4,5. The OMEGA effector TnpB is the putative ancestor of Cas12 and has RNA-guided endonuclease activity4,6. TnpB may also be the ancestor of the eukaryotic transposon-encoded Fanzor (Fz) proteins4,7, raising the possibility that eukaryotes are also equipped with CRISPR-Cas/OMEGA-like programmable RNA-guided endonucleases. Here we report the biochemical characterization of Fz, showing that it is an RNA-guided DNA endonuclease. We also show that Fz can be reprogrammed for human genome engineering applications. Finally, we resolved the structure of Spizellomyces punctatus Fz (SpuFz) at 2.7Ã… using cryogenic-electron microscopy, revealing the conservation of the core regions among Fz, TnpB and Cas12, despite diverse cognate RNA structures. Our results show that Fz is a eukaryotic OMEGA system, demonstrating that RNA-guided endonucleases are present in all three domains of life.";
 // Paper 3
 const paperAbstract3 = "This work introduces BioLORD, a new pre-training strategy for producing meaningful representations for clinical sentences and biomedical concepts. State-of-the-art methodologies operate by maximizing the similarity in representation of names referring to the same concept, and preventing collapse through contrastive learning. However, because biomedical names are not always self-explanatory, it sometimes results in non-semantic representations. BioLORD overcomes this issue by grounding its concept representations using definitions, as well as short descriptions derived from a multi-relational knowledge graph consisting of biomedical ontologies. Thanks to this grounding, our model produces more semantic concept representations that match more closely the hierarchical structure of ontologies. BioLORD establishes a new state of the art for text similarity on both clinical sentences (MedSTS) and biomedical concepts (MayoSRS).";


 return (
   <div className={styles.container}>
     <Head>
       <title>Let's Go Biotech - Read Paper</title>
       <meta name="description" content="Generated by Rohan in SF" />
       {/* <link rel="icon" href="/favicon.ico" /> */}
     </Head>

     <AppBar position="fixed" color="secondary">
      <Toolbar>
        <Typography variant="h6" component="div" sx={{ flexGrow: 1 }} className={styles.cursorPointer} onClick={goHome}>
          Let's Go Biotech 
        </Typography>
      </Toolbar>
    </AppBar>

    <Typography variant="body1" sx={{ mb: 2, fontSize: "2.0rem" }}>
      <span className={styles.readPaperHeader}><b>Enjoy Reading the Abstract</b></span>
    </Typography>
    

     {showPaper1 === 'false' && showPaper2 === "false" && showPaper3 === "false" && (
        <div>
           <Typography variant="body1" sx={{ mb: 2, fontSize: "3.0rem" }}>
              <span className={styles.lgbDescription}>You did not select any papers to read, go back and select one!</span>
           </Typography>
        </div>
     )}
      <div>

        {showPaper1 === 'true' && (
          <Card className={styles.abstractDisplay}>
            <CardContent>
              <Typography variant="body2" color="text.secondary">
                {paperAbstract1}
              </Typography>
            </CardContent>
          </Card>
        )}

        {showPaper2 === 'true' && (
          <Card className={styles.abstractDisplay}>
          <CardContent>
            <Typography variant="h5" component="div">Abstract</Typography>
            <Typography variant="body2" color="text.secondary">
              {paperAbstract2}
            </Typography>
          </CardContent>
          </Card>
        )}

        {showPaper3 === 'true' && (
          <Card className={styles.abstractDisplay}>
          <CardContent>
            <Typography variant="h5" component="div">Abstract</Typography>
            <Typography variant="body2" color="text.secondary">
              {paperAbstract3}
            </Typography>
          </CardContent>
          <CardContent>
          </CardContent>
        </Card>
        )}

    </div>

    <br />

        <Link
            className={styles.greenSquare} 
            href={{
                pathname: '/take-quiz',
                query: {
                  showPaper1: showPaper1,
                  showPaper2: showPaper2,
                  showPaper3: showPaper3
                }
            }}>
            <h2>Take Quiz</h2>
        </Link>

   

        <a href="/">
        <Typography variant="body1" sx={{ mb: 0.5 }}>
          &larr; <b>Select Different Paper</b>
      </Typography>
        </a>

   </div>
 )
}



